custom_components/moonraker/button.py

```diff
@@ -36,6 +36,31 @@ async def async_setup_entry(hass, entry, async_add_entities):
     coordinator = hass.data[DOMAIN][entry.entry_id]
     async_add_entities([MoonrakerButton(coordinator, entry, desc) for desc in BUTTONS])
 
+    await async_setup_macros(coordinator, entry, async_add_entities)
+
+
+async def async_setup_macros(coordinator, entry, async_add_entities):
```

Comment:
To be inline with style in Sensor, I would have this return a list of items to add and the add them in the above function.

Line 39:
macros = await async_setup_macros(coordinator, entry, async_add_entities)
async_add_entities([MoonrakerButton(coordinator, entry, desc) for desc in macros])

and then remove line 62

Reply:
FYI, this is mostly a nit pick

Reply:
I prefer that form actually, what if I make the sensors this way instead ?